docs/platform-specification.md

```diff
@@ -701,11 +739,53 @@ The file component of the port's path (e.g., `ttyACM0`) is available as the conf
 
 ### Upload using an external programmer
 
-**TODO...**<br> The platform.txt associated with the selected programmer will be used.
+The `program` action is triggered via the **Sketch > Upload Using Programmer** feature of the IDEs or
+[`arduino-cli upload --programmer <programmer ID>`](commands/arduino-cli_upload.md). This action is used to transfer a
+compiled sketch to a board using an external programmer.
+
+The **program.tool** property determines the tool to be used for this action. This property is typically defined for
+each programmer in [programmers.txt](#programmerstxt):
+
+    [......]
+    usbasp.program.tool=avrdude
+    [......]
+    arduinoasisp.program.tool=avrdude
+    [......]
+
+This action can use the same [upload verification preference system](#upload-verification) as the `upload` action, via
+the **program.verify** property.
+
+When using the Arduino IDE, if the selected programmer is from a different platform than the board, the `program` recipe
+of the programmer's platform is used without overrides from the properties of the [platform.txt](#platformtxt) of the
```

Comment:
these 3 "of" in the same paragraph sound a little off to me
"of the programmer's platform
"of the [platform.txt]"
"of the [board platform]"

Reply:
This is an excellent point. If I understand correctly, the problem is really about the nested "of"s in this part of the sentence:
>... the properties **of** the platform.txt **of** the board platform ...

Other than "from", which is not an option due to already being in use in the sentence, I didn't come up with a good replacement for the second "of". So my proposal is to replace the first of those two "of"s, as well as its parallel occurence earlier in the sentence, with "defined by":

>... the `program` recipe **defined by** the programmer's platform is used without overrides **from** the properties **defined by** the platform.txt **of** the board platform ...

What do you think @ubidefeo?

Reply:
now I like it 👌🏼

Reply:
I have now made the suggested change:
https://github.com/arduino/arduino-cli/compare/a7502c9b636f88ab2aa81d8eb4fa8869653a3f3d..8a5b4f5c6a08f78d0ba69277923dcc19478b49e4
Thanks!

This is a complex paragraph about something that has regularly caused confusion for platform authors over the years. Although the consistent behavior provided by Arduino CLI is a significant improvement, it did make the documentation more difficult due to the added requirement to explain the differences in the behavior of the tools. So I think that the improvement to the readability is especially important here.